ledger/catchpointtracker.go

```diff
@@ -414,7 +414,9 @@ func (ct *catchpointTracker) postCommit(ctx context.Context, dcc *deferredCommit
 	}
 
 	if dcc.isCatchpointRound && dcc.catchpointLabel != "" {
+		ct.catchpointsMu.Lock()
```

Comment:
This same lock is taken to update ct.roundDigest below, @algorandskiy do you think both ct.roundDigest and ct.lastCatchpointLabel need to be updated in the same locked section?

Reply:
Technically this is not necessary since we generate it every 10k rounds, and not every round.
But for consistency I agree, it is better to move this if statement under the lock below.
@Eric-Warehime could you please do that?